test/DotNetEnv.Tests/ParserTests.cs

```diff
@@ -238,8 +249,22 @@ public void ParseUnquotedValue ()
             // TODO: is it possible to get the system to recognize when a complete unicode char is present and start the next one then, without a space?
 //            Assert.Equal("日本", Parsers.UnquotedValue.Parse(@"\xe6\x97\xa5\xe6\x9c\xac"));
 
-            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("0\n1"));
-            Assert.Throws<ParseException>(() => Parsers.UnquotedValue.End().Parse("'"));
+            Assert.Equal("0", Parsers.UnquotedValue.Parse("0\n1").Value);   // value ends on linebreak
+            Assert.Equal("", Parsers.UnquotedValue.Parse("'").Value);  // unmatched singleQuote
+            Assert.Equal("", Parsers.UnquotedValue.Parse("'unmatched singleQuote").Value);
+            Assert.Equal("", Parsers.UnquotedValue.Parse("   'unmatched singleQuote with preceding whitespace").Value);
+            Assert.Equal("", Parsers.UnquotedValue.Parse("\t\t'unmatched singleQuote with preceding tabs").Value);
+            Assert.Equal("", Parsers.UnquotedValue.Parse("\"").Value);  // unmatched doubleQuote
+            Assert.Equal("", Parsers.UnquotedValue.Parse("\"unmatched doubleQuote").Value);
+            Assert.Equal("", Parsers.UnquotedValue.Parse("   \"unmatched doubleQuote with preceding whitespace").Value);
+            Assert.Equal("", Parsers.UnquotedValue.Parse("\t\t\"unmatched doubleQuote with preceding tabs").Value);
+
+            Assert.Equal("", Parsers.UnquotedValue.Parse("#").Value); // no value, empty comment
```

Comment:
This and the following examples about empty values followed by comments are all certainly excellent additions

Reply:
In fact, if you rewrite these first two in the style of `ParseEmpty` they pass already:
```
            Assert.Equal(kvp, Parsers.Empty.End().Parse("#"));
            Assert.Equal(kvp, Parsers.Empty.End().Parse("#commentOnly"));
```

And the following 3 tests already pass. (And have very similar matching examples in `ParseEmpty`)

Now I'm trying to remember why the `ParseEmpty` tests verify that the value is `null`, rather than `""`

Reply:
It is especially interesting to me that the last test case in `ParseEmpty` is effectively already this, but that test passes, whereas this one does not. I think that is the thread most worth pulling on. Because ideally, imo, any test in `ParseEmpty` returns `null`, should return `""` in `ParseUnquotedValue` and then a lot of these tests would just pass immediately...

Reply:
The `UnquotedValue`-Parser should parse something after an `EqualSign`, so only "Values" it is used within `Assignment`-Parser inside the `Value`-Parser).

The `Empty`-Parser should Parse a whole Line, an it is used only inside the `ParseDotenvFile`-Method directly. And it has to be null to be able to exclude these entries afterwards.

Back to the Tests:
I should definitely have a closer look here too I guess, I get some feeling that there is something I just missed while making the changes which has to be revealed (connected to the removed test in the other comments).
I will update the PR in a few days, giving more explanation and I guess some code changes.